### PR TITLE
WIP: New acceptance updates

### PR DESCRIPTION
**Pull request by wrightp**
New acceptance test suites for validating install.sh and install.ps1 scripts

### CODE REVIEW

**Comment by schisamo** [comment]
Can we either have these commits become part of https://github.com/chef/mixlib-install/pull/132 OR just have this PR target the `pw/package-router` branch.

**Comment by wrightp** [comment]
It can be part of 132, but I intentionally branched off 132 as not to muddy the waters if this PR wasn't quite ready in time. For now, I prefer that it is a separate PR to target package-router branch.

**Comment by schisamo** [comment]
@wrightp I appreciate you keeping #132 clean but let's have this PR target the `pw/package-router` so we are only reviewing a few new additional commits. That might mean closing this PR and opening a completely new one.

**Comment by wrightp** [comment]
I couldn't figure why you suggesting branching from the package-router branch since that's what I thought I did. Now I see that it branched from master (my bad). I will make it so!

File: acceptance/bourne/terraform/application.tf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    host         = "${self.private_ip}"
+    user         = "ubuntu"
+    private_key  = "${file("${var.connection_private_key}")}"
+    agent        = "${var.connection_agent}"
+  }
+
+  provisioner "file" {
+    source      = "${var.connection_private_key}"
+    destination = "/home/ubuntu/.ssh/id_rsa"
+  }
```
Comment (schisamo) [inline acceptance/bourne/terraform/application.tf]: What is this key used for?

Comment (wrightp) [inline acceptance/bourne/terraform/application.tf]: It's leftovers from the load client. Need to remove it.

File: acceptance/bourne/terraform/variables.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+variable "aws_region" {
+  type    = "string"
+  default = "us-west-2"
+}
+
+variable "aws_ami" {
+  description = "Base AMI for all nodes"
+
+  default = {
+    us-west-2 = "ami-06b94666"
```
Comment (wrightp) [inline acceptance/bourne/terraform/variables.tf]: cool!

File: acceptance/powershell/terraform/application.tf [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  key_name      = "es-infrastructure"
+
+  subnet_id         = "subnet-19ac017c" # Planet Releng Private Subnet
+  source_dest_check = false
+
+  vpc_security_group_ids = [
+    "sg-96274af3",
+  ]
+
+  # TODO: What needs to happen here?
```
Comment (wrightp) [inline acceptance/powershell/terraform/application.tf]: @schisamo This is here I'm blocked right now. I've tried all sorts of configurations. Will need to pair on this
```
    STDERR: Error applying plan:

    1 error(s) occurred:

    * unknown error Post http://10.194.11.185:5985/wsman: dial tcp 10.194.11.185:5985: getsockopt: operation timed out

    Terraform does not automatically rollback in the face of errors.
    Instead, your Terraform state file has been partially updated with
    any resources that successfully completed. Please address the error
    above and apply again to incrementally change your infrastructure.
```

Comment (schisamo) [inline acceptance/powershell/terraform/application.tf]: @wrightp Have you ensured `sg-96274af3` allows inbound WinRM traffic? You also might need to associate a public IP address with the instance. The `associate_public_ip_address` property/argument on the `aws_instance` resource should allow you to make this happen.

File: acceptance/bourne/terraform/application.tf [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    name = "block-device-mapping.volume-type"
+    values = ["gp2"]
+  }
+  filter {
+    name = "image-type"
+    values = ["machine"]
+  }
+}
+
+resource "aws_instance" "mixlib_install_sh" {
```
Comment (sersut) [inline acceptance/bourne/terraform/application.tf]: Optionally we can add our standard AWS tags.

Comment (wrightp) [inline acceptance/bourne/terraform/application.tf]: Added tags

**Comment by wrightp** [comment]
Just need to add inspecs

**Comment by wrightp** [comment]
I'm going to copy this acceptance dir directly into the package-router branch.

**Pull request closed without merging**